### ISSUE TITLE
Fail to fire torrent download for certain episodes: Cannot read property 'hashString' of undefined

### ISSUE THREAD

**Issue by marcelpaulo** [SchizoDuckie/DuckieTV#1084]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
Browse Action

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
807564b09d136a31e6228b7225c734c55eeba33f

**What is your Operating System (Windows, Mac, Linux, Android)**
Linux (Xubuntu 17.10)

**Describe the problem you are having and steps to reproduce if available**
DuckieTV doesn't start the download of _Krypton.S01E06.PROPER.HDTV.x264-KILLERS[ettv]_:
![image](https://user-images.githubusercontent.com/10074769/39305759-8cae7510-4934-11e8-9eca-33a6c6b03f4c.png)

It happens when downloading the torrent via any Jackett proxy: I tested the same torrent download with ETTV, 1337x, LimeTorrents, Demonoid and Torrentz2, and the error was exactly the same. The download through magnet links is fired OK. I also tried downloading other torrents via ETTV, 1337x, LimeTorrents, Demonoid and Torrentz2, and they're fired OK. It seems to be something specific with firing a torrent download for this particular torrent.

**Attach any DuckieTV statistics or Developer Console logs if available**
```
09:21:19.898 TorrentSearchEngines.js:257 Firing Torrent By data upload!  6496774 75C6912D3EAC47776D4A5ABC627FF6BDFD57E01B Krypton.S01E06.PROPER.HDTV.x264-KILLERS[ettv] null null
09:21:19.985 Transmission.js:190 Uncaught (in promise) TypeError: Cannot read property 'hashString' of undefined
    at Transmission.js:190
    at processQueue (angular.js:17169)
    at angular.js:17217
    at Scope.$digest (angular.js:18352)
    at Scope.$apply (angular.js:18649)
    at done (angular.js:12627)
    at completeRequest (app.authHttpBackend.js:160)
    at XMLHttpRequest.requestLoaded (app.authHttpBackend.js:101)
09:22:26.682 angular.js:14800 Possibly unhandled rejection: Canceled
```

In case it helps, I fired a magnet link download for _Krypton.S01E06.PROPER.HDTV.x264-KILLERS[ettv]_ and here's the torrent file that transmission ended up with:
[Krypton.S01E06.PROPER.HDTV.x264-KILLERS[ettv].zip](https://github.com/SchizoDuckie/DuckieTV/files/1951195/Krypton.S01E06.PROPER.HDTV.x264-KILLERS.ettv.zip)

**Comment by marcelpaulo** [comment]
Forgot to add: I'm using Jackett 0.8.920.0

**Comment by marcelpaulo** [comment]
If that helps to pinpoint the problem, the issue also happens with *Imposters.S02E04.iNTERNAL.HDTV.x264-W4F[ettv]*:
![image](https://user-images.githubusercontent.com/10074769/39363992-b4a18e6c-4a02-11e8-847e-1ce4944b97f1.png)
 with exactly the same error message on the console log:

```
Firing Torrent By data upload!  6625876 3784966B3176347D46510CAFA92F40CDC5B0EF88 Imposters.S02E04.iNTERNAL.HDTV.x264-W4F[ettv] null null
Transmission.js:190 Uncaught (in promise) TypeError: Cannot read property 'hashString' of undefined
    at Transmission.js:190
    at processQueue (angular.js:17169)
    at angular.js:17217
    at Scope.$digest (angular.js:18352)
    at Scope.$apply (angular.js:18649)
    at done (angular.js:12627)
    at completeRequest (app.authHttpBackend.js:160)
    at XMLHttpRequest.requestLoaded (app.authHttpBackend.js:101)
angular.js:14800 Possibly unhandled rejection: Canceled
```﻿

**Comment by garfield69** [comment]
I can't replicate this on my transmission/duckietv on my win7 development platform.
I tried both the examples you provided, and when my *Transmission 2.92* responds to duckietv that it has received the .torrent it was asked to start downloading, it returns back a *result* response that include the *torrent-added* details, which in turn includes the HashString that we are expecting to see.
But it appears that your Transmission is responding back to the .torrent request in a different way for some reason.
So what I have done is add some debugging console messages to that area of code, and in addition an attempt to handle the missing hashString, although I wont know what the downstream effect of this will be until you try it out for me.     

So what I'd like you to do please, is to grab the latest update, then repeat your test, and check the console log.
When you find messages similar to:
![image](https://user-images.githubusercontent.com/6933240/39385942-6b62ec18-4ac6-11e8-846e-74c0b0331edf.png)
i would like you to click on each arrow to expand the view of *arguments* and *torrents-added* so it looks similar to this:
![image](https://user-images.githubusercontent.com/6933240/39386010-b29f394c-4ac6-11e8-941a-199f06bc320e.png)
and then do a screen capture and post it here, and hopefully we shall see what is going on.

**Comment by marcelpaulo** [comment]
Oh, dear, tried both torrents, and this time they were both fired OK.

I was double-checking: I haven't restarted transmission today. I use the daemon, so it restarts only when I restart the laptop. Last night I was doing some tests, so I restarted the laptop a few times, but not today.

I've built transmission from source, I'm running transmission/transmission@444f15edf.

As neither of us can reproduce it, I propose we close the issue, and I'll make a point in always using the torrent download option from now onwards, and will reopen the issue with the extra debugging console output in case it happens again.
